Tenancy Bug - Getting 500 Response for Moving a Tenant and Renaming a Tenant

I am working within the new Traffic Portal and attempting to rename a tenant to a new name.  Also trying to move a tenant down to below a lower tenant.  Both updates return a 500 Internal Server Error.

Reproduce:
1. Open a tenant, change the name of the tenant, click update
2. Open a tenant, change the parent tenant, click update
@nir-sopher 
@mitchell852

@nir-sopher - any ideas? this appears to be the error

[2017-12-18 12:33:17,781] [ERROR] Check tenants relations - tenant Schema::Result::Tenant=HASH(0x7fc6ab796658) does not exists
Use of inherited AUTOLOAD for non-method API::Tenant::alert() is deprecated at /code/src/github.com/apache/incubator-trafficcontrol/traffic_ops/app/lib/API/Tenant.pm line 134.
[2017-12-18 12:33:17,782] [ERROR] Undefined subroutine &API::Tenant::alert called at /code/src/github.com/apache/incubator-trafficcontrol/traffic_ops/app/local/lib/perl5/Mojolicious.pm line 126.

--- a/traffic_ops/app/lib/API/Tenant.pm
+++ b/traffic_ops/app/lib/API/Tenant.pm
@@ -125,9 +125,9 @@ sub update {
 		return $self->alert("Root tenant cannot be in-active.");
 	}
 
-	if ($params->{parentId} != $tenant->parent) {
+	if ($params->{parentId} != $tenant->parent_id) {
 		#parent replacement
-		if (!$tenant_utils->is_tenant_resource_accessible($tenants_data, $tenant->parent)) {
+		if (!$tenant_utils->is_tenant_resource_accessible($tenants_data, $tenant->parent_id)) {
 			#Current owning tenant is not under user's tenancy
 			return $self>alert("Invalid parent tenant change. The current tenant parent is not avaialble for you to edit");
 		}

